pkg/ingress/model_build_listener_rules.go

```diff
@@ -72,6 +77,47 @@ func (t *defaultModelBuildTask) buildListenerRules(ctx context.Context, lsARN co
 	return nil
 }
 
+// sortIngressPaths will sort the paths following the strategy:
+// all exact match paths come first, no need to sort since exact match has to be unique
+// followed by prefix paths, sort by lengths - longer paths get precedence
+// followed by ImplementationSpecific paths or paths with no pathType specified, keep the original order
+func (t *defaultModelBuildTask) sortIngressPaths(paths []networking.HTTPIngressPath) ([]networking.HTTPIngressPath, error) {
+	exactPaths, prefixPaths, implementationSpecificPaths, err := t.classifyIngressPathsByType(paths)
+	if err != nil {
+		return nil, err
+	}
+	sortedPaths := exactPaths
+	sort.Slice(prefixPaths, func(i, j int) bool {
+		return len(prefixPaths[i].Path) > len(prefixPaths[j].Path)
+	})
+	sortedPaths = append(sortedPaths, prefixPaths...)
+	sortedPaths = append(sortedPaths, implementationSpecificPaths...)
+	return sortedPaths, nil
+}
+
+// classifyIngressPathsByType will classify the paths by type Exact, Prefix and ImplementationSpecific
+func (t *defaultModelBuildTask) classifyIngressPathsByType(paths []networking.HTTPIngressPath) ([]networking.HTTPIngressPath, []networking.HTTPIngressPath, []networking.HTTPIngressPath, error) {
+	var exactPaths []networking.HTTPIngressPath
+	var prefixPaths []networking.HTTPIngressPath
+	var implementationSpecificPaths []networking.HTTPIngressPath
+	for _, path := range paths {
+		if path.PathType != nil {
+			if *path.PathType == networking.PathTypeExact {
```

Comment:
use a switch statement instead of if-else

Reply:
Updated